### DISCUSSION TITLE
Storage Management not working

### DISCUSSION THREAD

**Discussion by DMarcusYoung** [supabase/supabase#15638] [Questions] [answered]
I'm following the Build a user management app with NextJS docs and I'm having trouble with the storage management section. I'm only made a few changes to the queries because my public table name is not profiles, but when I upload an avatar for the second time, the first image is still in the bucket.
This is what I ran, except with my actual URL and key
<img width="890" alt="Screenshot 2023-07-08 at 11 42 16 PM" src="https://github.com/supabase/supabase/assets/57845573/e0346f85-7c92-490a-a93a-5a3e9e65e8e5">
<img width="786" alt="Screenshot 2023-07-08 at 11 44 03 PM" src="https://github.com/supabase/supabase/assets/57845573/b912d6aa-bc71-4636-8786-280e9e358814">
<img width="636" alt="Screenshot 2023-07-08 at 11 44 29 PM" src="https://github.com/supabase/supabase/assets/57845573/5c56a52b-0647-45da-a4e7-9a3e25caf758">

Am I missing something?

Also a quick question
1. When it says "Note that it is not sufficient to delete the objects from the storage.objects table because that would orphan and leak the actual storage objects in the S3 backend"  is there a way to delete the actual storage object if it is orphaned?

Thanks for your help!

**Comment by GaryAustin1** [comment]
What does is still in the bucket mean?  You have two images or the image looks like the old image (caching is involved)?

Do you have a column name of avatar_url in your new table?  Any errors in the Postgres logs?  
You can also use `raise log 'value = %',variable;` to get more info in your functions if needed.  The output is also in the Postgres log.

1) No.  If it is orphaned it can't be deleted with any current API or SQL operation as far as I know.

**Reply by DMarcusYoung** [reply]
There are two images in the bucket, and yes I do have a column name of avatar_url in my influencer table. There is a 500 delete error in the Postgres logs, but I don't think it's from this. I could also be querying wrong.
<img width="1101" alt="Screenshot 2023-07-09 at 9 53 03 AM" src="https://github.com/supabase/supabase/assets/57845573/73c69f33-314b-47db-af0d-7326607f0da9">
Here's the message: 
`DELETE | 500 | 66.225.222.74 | 7e3bcdbf7a6ec459 | https://evsksteshxwikyndoptj.supabase.co/auth/v1/admin/users/735c5879-21cb-4ead-9b61-a36de42efe03 | supabase-api/d591a1d`

**Reply by DMarcusYoung** [reply, marked as answer]
The issue was I was using the anon key and not the service role key.
